### PR TITLE
prove more kernelc list examples

### PR DESCRIPTION
**Pull request by yilongli**

### CODE REVIEW

File: samples/kernelc/tests/list/reverse_spec.k [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2014 K Team. All Rights Reserved.
+require "../../kernelc.k"
+
+module REVERSE-SPEC
+  imports KERNELC
+
+rule [loop-inv]:
+  <struct>... STRUCT:Map ...</struct>
+  <k>
+    // if (x != NULL) { // TODO(YilongL): this rule is still provable after changing `while` to `if` (note that `LP1:Pointer => null`); is it a bug in ConstrainedTerm#matchImplies?
```
Comment (yilongli) [inline samples/kernelc/tests/list/reverse_spec.k:10]: @andreistefanescu can you confirm that it's indeed a bug in the implementation of checking implication?

**Comment by yilongli** [comment]
I planned to prove the `append`, `copy`, and `filter` examples which involve list segments but failed. So I am moving to those tree examples. @andreistefanescu please review. I will also prove `length_iterative` and `sum_iterative` after you have smt support for functions `len` and `sum`.

**Comment by yilongli** [comment]
Jenkins: retest this please

**Comment by andreistefanescu** [comment]
Looks good. I will look into the while vs if bug after I am done with the uninterpreted functions and list segments. Can you please make an issue and assign it to me? Thanks.

**Pull request merged**